From 82f907f4e468857c8e5b1e7fbf967cdd3a18a128 Mon Sep 17 00:00:00 2001
From: Mladen Milinkovic <maxrd2@smoothware.net>
Date: Sun, 7 Nov 2021 17:28:46 +0100
Subject: LinesItemDelegate: fix inline rich editor crash

Undo in inline editor would crash when current line gets deleted.
---
 src/gui/treeview/linesitemdelegate.cpp | 2 ++
 src/gui/treeview/linesmodel.cpp        | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/gui/treeview/linesitemdelegate.cpp b/src/gui/treeview/linesitemdelegate.cpp
index 1a691e4e..24fb0ae7 100644
--- a/src/gui/treeview/linesitemdelegate.cpp
+++ b/src/gui/treeview/linesitemdelegate.cpp
@@ -334,6 +334,8 @@ LinesItemDelegate::setEditorData(QWidget *editor, const QModelIndex &index) cons
 {
 	if(isRichDoc(index)) {
 		RichDocument *doc = index.data(Qt::EditRole).value<RichDocumentPtr>();
+		if(!doc)
+			return;
 		RichLineEdit *edit = static_cast<RichLineEdit *>(editor);
 		if(edit->document() != doc)
 			edit->setDocument(doc);
diff --git a/src/gui/treeview/linesmodel.cpp b/src/gui/treeview/linesmodel.cpp
index 3a871136..fbc16fc0 100644
--- a/src/gui/treeview/linesmodel.cpp
+++ b/src/gui/treeview/linesmodel.cpp
@@ -189,7 +189,7 @@ LinesModel::headerData(int section, Qt::Orientation orientation, int role) const
 QVariant
 LinesModel::data(const QModelIndex &index, int role) const
 {
-	if(!m_subtitle)
+	if(!m_subtitle || m_subtitle->count() <= index.row())
 		return QVariant();
 
 	SubtitleLine *line = m_subtitle->at(index.row());
-- 
2.38.1

